src/simplicial-hott/08-covariant.rzk.md

```diff
@@ -1682,3 +1683,31 @@ commuting with the contravariant lifts.
       is-contravariant-representable-is-segal A is-segal-A)
 
 ```
+## Closure properties of covariance
+
+```rzk title="RS17, Theorem 8.30"
+#def weak-extensionality 
+  ( A : U )
+  : U :=
+   (C : A → U) →  (f : (a : A) → is-contr (C a) ) → (is-contr ( (a : A) → C a )) 
+
+#def locally-covariant-implies-covariant-dependent-types
+   (A B : U)
+   (C : A → B → U)
+   (is-locally-covariant : (b : B) → is-covariant A ( \ a → C a b ) )
+  (we : weak-extensionality B)
+  : is-covariant A ( \ a → (( b : B ) → (C a b)) ) 
+  := 
+    ( first ( has-unique-fixed-domain-lifts-iff-is-covariant A ( \ a → ( b : B ) → (C a b) )) ) 
+        ( \ x y f g →  
+           equiv-with-contractible-codomain-implies-contractible-domain
+           ( (t : Δ¹) → ((b : B) → C (f t) b) [  t ≡ 0₂ ↦ g ])   
+           ( (b : B) → (t : Δ¹) → C (f t) b [ t ≡ 0₂ ↦ g b]) 
+           ( flip-ext-fun 2 Δ¹ (\ t → t ≡ 0₂) B ( \ t →  C (f t)) ( \ t → g)) 
+          (we 
+            ( \ b → ( (t : Δ¹) → C (f t) b [ t ≡ 0₂ ↦ g b] ) ) 
+            ( \ b → ( (( second ( has-unique-fixed-domain-lifts-iff-is-covariant A ( \ a  → (C a b) )) ) (is-locally-covariant b)) x y f (g b) ) )
+            )
+        ) 
```

Comment:
Here, I assume it'd be easier (and more helpful) to talk personally to me or @emilyriehl about the [Code style](https://rzk-lang.github.io/sHoTT/STYLEGUIDE/). But if you want you can try adjusting this yourself before we can meet :)

Reply:
Thanks for the link. I'll give it a shot based on the link and then make sure to get feedback afterwards when we meet in the coming days.